examples/microwave-oven-app/linux/main.cpp

```diff
@@ -0,0 +1,50 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include "microwave-oven-device.h"
+#include <AppMain.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+
+#define MICROWAVE_OVEN_ENDPOINT 1
+
+MicrowaveOvenDevice * gMicrowaveOvenDevice = nullptr;
```

Comment:
how about using a std::unique_ptr instead of raw pointers?

Reply:
Yes, using std::unique_ptr is better, so that is no need to delete objects manually, and is suitable for some plaform which without Shutdown() function.